### PR TITLE
InitializePlugins: verify that callback init/1 of the plugin module i…

### PR DESCRIPTION
**Pull request by redmaner**
In addition to PR #929, the `Code.ensure_loaded?/1` function only makes sure that the plugin is loaded but does check if the plugin module also exports the required callback function `init/1`. This PR adds an extra check to verify that the the required callback function `init/1` is exported in the plugin module. If it is not exported we can safely assume the module doesn't implement the required behavior. 

I have tested the commit with one of our own applications that uses `CredoSonarqube` as a plugin, which works as expected. I also tested it with another module that doesn't have an `init/1` function, and with this false plugin the error is raised properly.

### CODE REVIEW

File: lib/credo/execution/task/initialize_plugins.ex [hunk 1]
```diff
@@ -12,19 +12,19 @@ defmodule Credo.Execution.Task.InitializePlugins do
...
     else
-      Execution.halt(
-        exec,
-        "Plugin module `#{Credo.Code.Module.name(mod)}` is not available and could not be initialized."
-      )
+      false ->
+        Execution.halt(
+          exec,
+          "Plugin module `#{Credo.Code.Module.name(mod)}` is not available and could not be initialized or is not a valid credo plugin."
+        )
```
Comment (rrrene) [inline lib/credo/execution/task/initialize_plugins.ex:27]: Great 👍  We need a nested `if` here so we can give a concise error message. "It is this or this" is not a great user experience.

Comment (redmaner) [reply lib/credo/execution/task/initialize_plugins.ex]: Thanks for your review, good point! I've updated the PR to include separation of error messages for a better user experience

Comment (rrrene) [reply lib/credo/execution/task/initialize_plugins.ex]: Hi, please update the PR according to my suggestion. You forced and unnecessary usage of `with` makes the code less readable with every iteration.

Comment (redmaner) [reply lib/credo/execution/task/initialize_plugins.ex]: Done

**Comment by rrrene** [comment]
@redmaner Thx! :+1:

**Pull request merged**